README.md

```diff
@@ -85,6 +85,102 @@ module.exports = {
 };
 ```
 
+## Multiple themes (dark/light mode)
+
+Because Shiki generates themes at build time, client-side theme switching
+support is not built in. There are two popular options for supporting something
+like Dark Mode with Shiki. See the
+[Shiki docs](https://github.com/shikijs/shiki/blob/main/docs/themes.md#dark-mode-support)
+for more info.
+
+#### 1. Use the "css-variables" theme (Shiki version `0.9.9` and above).
+
+This gives you access to CSS variable styling, which you can control across Dark
+and Light mode.
+
+Note that this client-side theme is less granular than most other supported
+VS Code themes. Also, be aware that this will generate unstyled code if you do
+not define these CSS variables somewhere else on your page:
+
+```css
+<style>
+  :root {
+    --shiki-color-text: rgb(248, 248, 242);
+    --shiki-color-background: rgb(13 13 15);
+    --shiki-token-constant: rgb(102, 217, 239);
+    --shiki-token-string: rgb(230, 219, 116);
+    --shiki-token-comment: rgb(93,93, 95);
+    --shiki-token-keyword: rgb(249, 38, 114);
+    --shiki-token-parameter: rgb(230, 219, 116);
+    --shiki-token-function: rgb(166, 226, 46);
+    --shiki-token-string-expression: rgb(230, 219, 116);
+    --shiki-token-punctuation: rgb(230, 219, 116);
+    --shiki-token-link: rgb(174, 129, 255);
+  }
+</style>
+```
+
+#### 2. Load multiple themes
+
+This will render duplicate code blocks for each theme. You can then hide the
+other blocks with CSS.
+
+Pass an array of themes to `shikiOptions.themes`:
+
+```js
+  shikiOptions: {
+    // Link to your VS Code theme JSON file
+    themes: [
+      JSON.parse(
+        fs.readFileSync(require.resolve('./themes/my-theme.json'), 'utf-8')
+      ),
+      JSON.parse(
+        fs.readFileSync(require.resolve('./themes/my-other-theme.json'), 'utf-8')
+      ),
+    ],
+  },
+```
+
+The `code` elements and the inline code `<span data-mdx-pretty-code>` wrappers
+will have a class of `theme.name` (spaces will be removed, but it's otherwise up
+to you to make sure this is valid as a CSS class; default themes should already
+have a kebab-cased name).
```

Comment:
Should we maybe make them specify a `type` in the themes array? Like:

```js
themes: [
  {name: 'light', json: JSON.parse(...)},
  {name: 'dark', json: JSON.parse(...)}
]
```

Feel like this would be more explicit.

I'd also like this to be the first/recommended method, with the less granular CSS variables version maybe collapsed in `<details>` beneath it. IMO the granular highlighting using any VS Code was the primary reason why I bothered creating this 😅

Reply:
I did wonder this, probably makes more sense.
alternatively, make it an object?
```
themes: {
    dark: 'min-dark',
    light: 'min-light'
}
```

Any thoughts on the classes? Is it too verbose? With the above light/dark keys or name it would be:  `min-light mdx-pretty-code-theme-light` | `min-dark mdx-pretty-code-theme-dark`

Good call on the `<details>`.

Reply:
I like the object syntax. I wonder if it's necessary to pluralize the key and just overload `theme` instead? 

I think using an attribute would work fine: `data-theme="light"`.

Reply:
re: pluralized key, I don't think you could easily detect the difference between an actual (single) theme object and multiple themes in an object?

Reply:
You could perform duck typing and check for `tokenColors` — if that exists, it's the JSON theme object, vs. an object of theme names